core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -455,13 +516,38 @@ public TableMetadata buildReplacement(Schema updatedSchema, PartitionSpec update
     return new TableMetadata(ops, null, uuid, location,
         System.currentTimeMillis(), nextLastColumnId.get(), freshSchema,
         specId, builder.build(), ImmutableMap.copyOf(newProperties),
-        -1, snapshots, ImmutableList.of());
+        -1, snapshots, ImmutableList.of(), addPreviousMetadata(file, lastUpdatedMillis, newProperties));
   }
 
   public TableMetadata updateLocation(String newLocation) {
     return new TableMetadata(ops, null, uuid, newLocation,
         System.currentTimeMillis(), lastColumnId, schema, defaultSpecId, specs, properties,
-        currentSnapshotId, snapshots, snapshotLog);
+        currentSnapshotId, snapshots, snapshotLog, addPreviousMetadata(file, lastUpdatedMillis));
+  }
+
+  private List<MetadataLogEntry> addPreviousMetadata(InputFile previousFile, long timestampMillis) {
+    return addPreviousMetadata(previousFile, timestampMillis, properties);
+  }
+
+  private List<MetadataLogEntry> addPreviousMetadata(InputFile previousFile, long timestampMillis,
+                                                     Map<String, String> updatedProperties) {
+    if (previousFile == null) {
+      return previousMetadata;
+    }
+
+    int maxSize = PropertyUtil.propertyAsInt(updatedProperties,
```

Comment:
This should ensure that the property value is at least 1. How about adding `Math.max(1, ...)`.